### PR TITLE
TN-2121 reformat summary reports

### PR DESCRIPTION
**Pull request by pbugni**
Site summary request report now includes dates and patient details as requested in TN-2121

### CODE REVIEW

File: portal/migrations/versions/9c6788e6db2f_.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        sys_user.update_roles(role_list=admin_only, acting_user=sys_user)
+
+    # CYA - give the system account a decent, random password
+    password = (''.join([random.choice(list(
+        '123456789qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM'))
+        for x in range(32)]))
+    pw_hash = generate_password_hash(password)
+    # pluck the hash algorithm from the hash, like all others in the db
+    i = pw_hash.find('$')
+    sys_user.password = pw_hash[i:]
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/migrations/versions/9c6788e6db2f_.py]: I think `user.password` ([managed by `flask_user`](https://github.com/uwcirg/Flask-User/blob/0.6.21.2/flask_user/passwords.py#L16-L27)) may have a different format (at least in the DB field, not sure if the model adds some accessor magic)?
```
$ dc exec db psql --username postgres --dbname portaldb -c 'select password from users limit 10'
                           password                           
--------------------------------------------------------------
 
 $2b$12$dFK.Ll/RQ01/lLgF.XhAZeGdp/jxfF1CigguDzlwbEP7TsQyb7noe
 $2b$12$qWQyfqOT0Y5T05ktfLAiAurgP0xQt5/Tef77qQWfjLm3VGJWAEbxa
 $2b$12$J.CIj3sizXCHEoYhPWuyAeGgOn09gp081oFa17ceTDEIVWQgZijui
 $2b$12$dFK.Ll/RQ01/lLgF.XhAZeGdp/jxfF1CigguDzlwbEP7TsQyb7noe
 
 $2b$12$AzJ/eng3yN1LQo.j5hxed..btYfZFKbEuITXl9Ce5D5n1zqvaV7Xy
 $2b$12$4j1tiAyEt4Zty4cMf/XzB.nj3bblhzpYbhorNdnNZfsGm0xrepeXK
```

File: portal/migrations/versions/9c6788e6db2f_.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def upgrade():
+    sys_user = User.query.filter_by(email='__system__').one()
+    if not(len(sys_user.roles) == 1 and
+           sys_user.has_role(ROLE.ADMIN.value)):
+        admin_only = Role.query.filter(Role.name == ROLE.ADMIN.value).all()
+        sys_user.update_roles(role_list=admin_only, acting_user=sys_user)
+
+    # CYA - give the system account a decent, random password
```
Comment (ivan-c) [inline portal/migrations/versions/9c6788e6db2f_.py]: It's more consistent/secure to directly set the hash to an empty or impossible value. No hash function will ever return an empty or truncated string.

Comment (pbugni) [reply portal/migrations/versions/9c6788e6db2f_.py]: can you give me an impossible value example? that's an attractive solution.  leaving blank has me paranoid

File: portal/migrations/versions/9c6788e6db2f_.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+revision = '9c6788e6db2f'
+down_revision = 'c242e22f5a47'
+
+
+def upgrade():
+    sys_user = User.query.filter_by(email='__system__').one()
+    if not(len(sys_user.roles) == 1 and
+           sys_user.has_role(ROLE.ADMIN.value)):
+        admin_only = Role.query.filter(Role.name == ROLE.ADMIN.value).all()
+        sys_user.update_roles(role_list=admin_only, acting_user=sys_user)
```
[the remark this answers is not in the document]
Comment (pbugni) [reply portal/migrations/versions/9c6788e6db2f_.py]: I'd overlooked the fresh DB situation, thinking we always have that defined (at least everywhere I looked).
Hmm - we don't yet have a persistence file for users.  That would be a better long term solution for this.  Do you feel it's necessary at this time?  It'll fail in a very noisy manner (celery will raise stack) if there's no `__system__` user.

[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/migrations/versions/9c6788e6db2f_.py]: >Do you feel it's necessary at this time?

I'm fine punting, unless it breaks our builds, [which build, do a full fresh deploy of the portal and healthcheck](https://github.com/uwcirg/truenth-portal/blob/develop/tox.ini#L61-L77).

[the remark this answers is not in the document]
Comment (ivan-c) [reply portal/migrations/versions/9c6788e6db2f_.py]: Another short-term fix could be to create the `__system__` user via CLI as part of docker, after we call `flask sync`

File: portal/migrations/versions/9c6788e6db2f_.py (hunk 2 above)
Comment (ivan-c) [reply portal/migrations/versions/9c6788e6db2f_.py]: something simple and self-explanatory would be ideal eg `impossiblepasswordhash` in place of the hashed password

**Pull request merged**